src/uwtools/config.py

```diff
@@ -752,3 +753,35 @@ def create_config_obj(
                     raise ValueError(err_msg)
                 # Dump to file:
                 dump_method(path=outfile, cfg=config_obj)
+
+
+def log_and_error(msg: str, log: Logger) -> None:
+    """
+    Will log a user-provided error message and raise a UWConfigError with the same message.
+    """
+    log.error(msg)
+    raise UWConfigError(msg)
+
+
+def print_config_section(config: dict, section_path: List[str], log: Logger) -> None:
+    """
+    Descends into the config via the given section keys, then prints the contents of the located
+    subtree as key=value pairs, one per line.
+    """
+    keys = []
+    for section in section_path:
+        keys.append(section)
+        current_path = " -> ".join(keys)
+        try:
+            subconfig = config[section]
+        except KeyError:
+            log_and_error(f"Bad config path: {current_path}", log)
+        if not isinstance(subconfig, dict):
+            log_and_error(f"Value at {current_path} must be a dictionary", log)
+        config = subconfig
+    output_lines = []
+    for key, value in config.items():
+        if type(value) not in (bool, float, int, list, str):
+            log_and_error(f"Non-dictionary value {value} found at {current_path}", log)
```

Comment:
Maybe I've lost the plot, but I think that by the time we reach this `for` loop, we know that `config` is a `dict`, and now we want to iterate over its key-value pairs, ensure that all values are scalars, add them to a list of key/val pairs to output if so, abort if not. If that's right, 1. Should we remove `list` from the tuple of acceptable (i.e. scalar) types, and 2. Complain about a `Non-scalar` value instead of a `Non-dictionary` value?

Reply:
Please ignore, I commented on old code. Resolving...

Reply:
That was me who lost the plot. I already corrected it in the last push.